main/simulator.cpp

```diff
@@ -1 +1,78 @@
 #include "main/simulator.h"
+
+std::vector<std::unique_ptr<objects::EnergyCollector>>
+buildCollectors(const ModelInterface &model, int numCollectors) {
+
+  if (model.empty()) {
+    throw std::invalid_argument("Empty model");
+  }
+  if (numCollectors < 4) {
+    std::stringstream ss;
+    ss << "numCollectors: " << numCollectors << " is less then 4";
+    throw std::invalid_argument(ss.str());
+  }
+
+  // We distribute collectors along 4 quarter circles above the model, so
+  // |numCollectors| has to be divisible by 4. Or |numCollectors|-1 has to be
+  // divisible by 4 (if it is odd), in which case there is one more collector
+  // placed right above the model.
+  if (numCollectors % 4 != 0 && (numCollectors - 1) % 4 != 0) {
+    std::stringstream ss;
+    ss << "numCollectors or numCollectors-1 has to be divisible by 4, got "
+       << "numCollectors = " << numCollectors;
+    throw std::invalid_argument(ss.str());
+  }
+
+  // Minimum distance from Vec3(0, 0, 0) that each collector must meet.
+  // Value of the minDistance can't be less then 4 [m]
+  float minDistance =
```

Comment:
1. minDistance -> collectorSphereRadius (or something better): minDistance suggest that actual distance can be larger than this, but in practice it cannot. it must be exactly this.

2. change formula to be more aligned with what you have in specification -- about being at least 4 -- just write "at least 4" and not some workaround so that it turns out to be 4, so:

```std::max(4, 4 * std::max(height, sideSize()));```

Then there is no need to comment why do you need to add 1 to the max below.

3. is the <float> actually needed here?